upgrades/upgrade_testing_framework/steps/step_load_test_config.py

```diff
@@ -0,0 +1,24 @@
+import upgrade_testing_framework.cluster_management.docker_framework_client as dfc
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+import upgrade_testing_framework.core.test_config_wrangling as tcw
+
+class LoadTestConfig(FrameworkStep):
+    """
+    This step confirms the test configuration is valid and loads it into the Framework's state
+    """
+
+    def _run(self):
+        # Get the state we need
+        test_config_path = self._get_state_value("test_config_path") 
+
+        # Begin the step body
+        self.logger.info("Loading test config file...")
+        try:
+            test_config = tcw.load_test_config(test_config_path)
+        except Exception as exception:
+            self.fail(f"Unable to load test config file.  Details: {str(exception)}", exception)
+        self.logger.info("Loaded test config file successfully")
+        
+        # Update our state
+        self._set_state_value("test_config", test_config)
```

Comment:
No issues, but points for discussion:
1. What's your thinking between putting the test_config as a full object in the state vs. unpacking it to multiple (in the future) top-level items?
2. Do you think of the state as read/append-only?  This is a case where it feels not-unreasonable to replace the file path with the config itself, but I'm I think I have been overall assuming it's append-only -- could be worth formalizing our expectations for how steps interact with the state.

Reply:
Good questions.

1. I don't have strong opinions about this at the moment; as long as its easy to write/maintain, I'm down for whatever.  That said, it feels a bit better to have separate objects for each highest-level `test_config.json` key just to make the individual unit tests simpler.
2. Yeah, I'm thinking about it as append-only.  That would be good to call out somewhere explicitly, per your comment.